src/app/logic/quiz.js

```diff
@@ -0,0 +1,109 @@
+const axios = require('axios');
+
+const gameInfo = {};
+
+const fetchItems = (api) => {
+  axios
+    .get(api)
+    .then((item) => item.data)
+    .then((item) => {
+      gameInfo.questionScope.push(...item.results);
+      if (item.next !== null) {
+        fetchItems(item.next);
+      }
+      if (item.count === gameInfo.questionScope.length) {
+        gameInfo.questionScope = gameInfo.questionScope.map(
+          (value) => ({
+            name: value.name,
+            idx: value.url.split('/')[5],
+          }),
+        );
+        localStorage.setItem(
+          gameInfo.mode,
+          JSON.stringify(gameInfo.questionScope),
+        );
+      }
+    })
+    .catch((err) => {
+      throw Error(`fetchError: ${err}`);
+    });
+};
+
+const fetchQuestionScope = () => {
+  gameInfo.questionScope = [];
+  fetchItems(`${gameInfo.apiUrl}/${gameInfo.mode}/`);
+};
+
+const pickRandomId = () => {
+  let result;
+  while (result === undefined) {
+    result = Math.ceil(Math.random() * gameInfo.questionScope.length);
+    if (result === gameInfo.questionScope.length) {
+      result = 0;
+    }
+    if (gameInfo.questionScope[result] === undefined) {
+      result = undefined;
+    }
+  }
+  return result;
+};
+
+const getUrlData = (id, question) => {
+  fetch(`/static/assets/img/modes/${gameInfo.mode}/${id}.jpg`)
+    .then((res) => res.blob())
+    .then((blob) => {
+      const reader = new FileReader();
+      reader.onload = () => {
+        question.image = reader.result;
+      };
+      reader.readAsDataURL(blob);
+    })
+    .catch(() => {
+      question.err = 'imageError';
+    });
+};
+
+const generateQuestion = () => {
+  const question = {};
+  if (gameInfo.questionScope.length === 0) {
+    question.err = 'modeOrDataError';
+    return question;
+  }
+  const answersIDs = [];
+  while (answersIDs.length !== 4) {
+    const id = pickRandomId();
+    if (!answersIDs.some((fId) => fId === id)) {
+      answersIDs.push(id);
+    }
+  }
+  let correctIdIdxFromAnswers = Math.ceil(
+    Math.random() * answersIDs.length,
+  );
+  if (correctIdIdxFromAnswers === 4) {
+    correctIdIdxFromAnswers = 0;
+  }
+  const rightAnswer = answersIDs[correctIdIdxFromAnswers];
+  const rightAnswerApiDataId =
+    gameInfo.questionScope[rightAnswer].idx;
+  getUrlData(rightAnswerApiDataId, question);
+  question.answers = answersIDs.map(
+    (id) => gameInfo.questionScope[id].name,
+  );
+  question.rightAnswer = gameInfo.questionScope[rightAnswer].name;
+  return question;
+};
+
+export const initGame = (mode, url) => {
+  gameInfo.apiUrl =
+    url || process.env.SW_API_BASE_URL || 'https://swapi.dev/api';
+  gameInfo.mode = mode;
+  if (localStorage.getItem(mode) !== null) {
+    gameInfo.questionScope = JSON.parse(localStorage.getItem(mode));
+  } else {
+    fetchQuestionScope();
+  }
```

Comment:
```suggestion
  const cached = localStorage.getItem(mode) || 0
  if (cached) {
    gameInfo.questionScope = JSON.parse(localStorage.getItem(mode));
    return
  } 
  fetchQuestionScope();
 
```

Reply:
Changed localStorage verification as you suggested